weblab/entities/forms.py

```diff
@@ -31,6 +36,29 @@ def save(self, **kwargs):
     def entity_type(self):
         return self._meta.model.entity_type
 
+class EntityRenameForm(UserKwargModelFormMixin, forms.ModelForm):
+    """Used for renaming an existing entity."""
+
+    def clean_name(self):
+        name = self.cleaned_data['name']
+        if self._meta.model.objects.filter(name=name).exists():
```

Comment:
I thought we might need a `, author=self.user` argument in the filter too, but `EntityForm` doesn't have this. So I think there may be a missing test case for creation too, that checks it's OK for 2 users to have a model with the same name!

Reply:
sounds like we do need to filter by username in the form so I have added `author=self.user` as a filter option - is this correct?

Reply:
interestingly I now get a database integrity error thrown, when I try to a rename model to one that exists. Not sure if I need to trap this, was just expecting the message from the form
 
![image](https://user-images.githubusercontent.com/3098326/94008782-7735e380-fd9b-11ea-9246-977ba3939c3b.png)

Reply:
Interesting. `Entity` definitely has `unique_together = ('entity_type', 'name', 'author')` so should allow same name different author. And it looks like from the error it's complaining that the author also matches. So as you say I'd expect the form validation to catch this before it hits the DB. Could you push/paste your work-in-progress so I can take a look?

Reply:
The test looks OK, so it might be something in this file that's wrong, but I don't think you've pushed any changes here?